### PR TITLE
feat: Add --draft flag for draft pull requests

### LINKED ISSUE #47: Add option to create new PRs in draft mode
It would be great to have an option that make git-xargs open the PRs in draft mode. Sadly the GitHub Rest-API does not provide a resource to change the draft-state after creating a PR, so the only possibility is to do this at creation time.

### PR DESCRIPTION
**Pull request by ryanwholey**
Hey folks, thanks for the neat tool!

My team is interested in using the GitHub drafts feature so I've implemented a `--draft` flag here.

I played around with a few tests but I didn't find one that felt like it provided meaningful coverage. What we want I think is a test where we check that the PR submitted to [`client.PullRequests.Create`](https://github.com/gruntwork-io/git-xargs/blob/7b8ec9c37a780a0d163010b761074d09ed33b224/repository/repo-operations.go#L448) has the has `Draft: true`, which would require some storage structure on the mock used in perhaps another test like [`TestProcessRepo`](https://github.com/gruntwork-io/git-xargs/blob/b55137d2a13d042ca5930c1b1e1dcca8b24230b9/repository/process_test.go#L16). This did seem like a lot for a flag, so wanted to see what you thought before adding it. Open to any suggestions for testing approaches!

Also looks like my editor removed a couple extra spaces, please let me know if you'd like them reverted.   

closes #47

### CODE REVIEW

**Comment by ryanwholey** [comment]
Hey @brikis98 @zackproser! Any interest in the changes proposed here? Thanks!

**Comment by zackproser** [comment]
Hi @ryanwholey, 

Thanks so much for contributing to `git-xargs` and apologies for the delay in getting back to you. 

First off, this looks like a great start and I do believe we'd be interested in this functionality. There are a couple of things we'd probably want to include before this is ready to merge, however: 
1. Error handling for when draft pull requests are not supported. For example, I have a personal Github org that I use for testing. It is currently comprised of only private repos, [which currently do not support draft PRs according to Github's docs](https://github.blog/2019-02-14-introducing-draft-pull-requests/#get-started) - and, when testing this locally against my test org, if I pass the `--draft` flag and the `--loglevel DEBUG` flag, I can see HTTP 422 errors being returned as a result. We should probably handle this error specifically
2. In order for operators to be notified of any pull requests that could not be opened as drafts in the final run report, we'll need to wire up a new event type in `stats/stats.go` and add it to the slice of all events in that same file. This way, when someone attempts to open draft pull requests on a repo that does not support them, they'll get a detailed breakdown of exactly which repos had this issue, so it will be easier to troubleshoot. 
3. The additional test coverage, as you suggested. This would be ideal to have in place before merging. However, if it's not something you have the interest or bandwidth to implement, we could do it on our end, but this may take a bit longer.

**Comment by ryanwholey** [comment]
@zackproser all good, thank you for the direction! I'll dive in and update soon

**Comment by ryanwholey** [comment]
Hi @zackproser, finally getting back around to this!

I've added handling for 422's under the event type `RepoNotCompatibleWithPullConfig`. Looks like the GitHub client doesn't export more specified errors here for something like draft incompatibility. Let me know if you think we need some more targeted copy shown to the user.

Regarding updating the testing framework here, I don't believe I'll have time to get to this any time soon. Please let me know if you'd like to see some other form of validation here with regard to these changes.

**Comment by zackproser** [comment]
> Hi @zackproser, finally getting back around to this!
> 
> I've added handling for 422's under the event type `RepoNotCompatibleWithPullConfig`. Looks like the GitHub client doesn't export more specified errors here for something like draft incompatibility. Let me know if you think we need some more targeted copy shown to the user.
> 
> Regarding updating the testing framework here, I don't believe I'll have time to get to this any time soon. Please let me know if you'd like to see some other form of validation here with regard to these changes.

Hi @ryanwholey, 

Awesome - thanks again for implementing this! I was able to get draft PRs opened using your branch against my test org, so that's great. 

I have a couple of nits and thoughts around how to best describe the draft PR activity in the final run report that gets rendered for operators. 

I'm going to keep playing around with this and leave some feedback on your PR and possibly also push some commits there.

File: README.md [hunk 1]
```diff
@@ -377,6 +377,8 @@ echo "gruntwork-io/terragrunt gruntwork-io/terratest" | git-xargs \
...
 | `--max-concurrent-repos` | Limits the number of concurrent processed repositories. This is only useful if you encounter issues and need throttling when running on a very large number of repos. Default is `0` (Unlimited)                                                                                                                                                                                                                              | Integer | No       |
+| `--draft` | Whether to open pull requests in draft mode. Draft pull requests are available for public GitHub repositories and private repositories in GitHub tiered accounts. See [Draft Pull Requests](https://docs.github.com/en/github/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/about-pull-requests#draft-pull-requests) for more details.  | Boolean | No |
```
Comment (zackproser) [inline README.md:380]: :+1: Doc entry looks good!

File: printer/printer.go [hunk 2]
```diff
@@ -93,4 +103,16 @@ func PrintRepoReport(allEvents []types.AnnotatedEvent, runReport *types.RunRepor
 		fmt.Println()
 
 	}
+
```
Comment (zackproser) [inline printer/printer.go:106]: In order to keep with the spirit of making it really easy to understand exactly what happened with each run, I decided to treat pull requests and draft pull requests as two separate entities for reporting purposes. 

The result is more explicit messaging in the run report about which PRs were drafts and which draft PRs couldn't be opened due to repo incompatibility. If you don't pass the `--draft` flag, you'll get pull requests, and if you do pass it, you end up with draft pull requests.

**Comment by zackproser** [comment]
I opened https://github.com/gruntwork-io/git-xargs/issues/61 to track the implementation of additional test coverage for draft PRs.

**Review by infraredgirl — APPROVED** [review]

**Comment by zackproser** [comment]
Thanks for the review! Going to merge this in now, as tests also passed locally.

**Pull request merged**